client/modules/IDE/components/Searchbar/Searchbar.jsx

```diff
@@ -1,82 +1,78 @@
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import { throttle } from 'lodash';
 import { withTranslation } from 'react-i18next';
 import i18next from 'i18next';
 import SearchIcon from '../../../../images/magnifyingglass.svg';
 
-class Searchbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      searchValue: this.props.searchTerm
-    };
-    this.throttledSearchChange = throttle(this.searchChange, 500);
-  }
+const Searchbar = ({
+  searchTerm,
+  setSearchTerm,
+  resetSearchTerm,
+  searchLabel,
+  t
+}) => {
+  const [searchValue, setSearchValue] = useState(searchTerm);
 
-  componentWillUnmount() {
-    this.props.resetSearchTerm();
-  }
+  const searchChange = () => {
+    setSearchTerm(searchValue.trim());
+  };
+
+  const throttledSearchChange = throttle(searchChange, 500);
+
+  useEffect(
+    () => () => {
+      resetSearchTerm();
+    },
+    [resetSearchTerm]
+  );
```

Comment:
This is technically correct but confusing to read, as it's not clear that it's a cleanup function.  But I know that prettier/eslint will complain if you have a `return` instead of an arrow function.  Hmmm...

Reply:
just replace () with return  but even without replacing it eslint/prettier is working good .

useEffect(() => {
    return () => {
      resetSearchTerm();
    };
  }, [resetSearchTerm]);